core/View.php

```diff
@@ -458,7 +459,27 @@ public static function singleReport($title, $reportHtml)
     private function shouldPropagateTokenAuthInAjaxRequests()
     {
         $generalConfig = Config::getInstance()->General;
-        return Common::getRequestVar('module', false) == 'Widgetize' || $generalConfig['enable_framed_pages'] == '1';
+        return Common::getRequestVar('module', false) == 'Widgetize' ||
+            $generalConfig['enable_framed_pages'] == '1' ||
+            $this->validTokenAuthInUrl();
+    }
+
+    /**
+     * @param bool $return the token_auth $_GET variable
+     * @return bool|string
+     * @throws Exception
+     */
+    private function validTokenAuthInUrl(bool $return = false)
+    {
+        $tokenAuth = Common::getRequestVar('token_auth', '', 'string', $_GET);
+        if ($tokenAuth) {
```

Comment:
@geekdenz here we should be possible to simply do `return $tokenAuth && $tokenAuth == Piwik::getCurrentUserTokenAuth()`. The logic below might not work eg SessionAuth should not be used when the API is involved and it may miss some other needed logic like brute force detection etc.

Also the boolean parameter isn't needed?